src/python/pants/option/option_value_container.py

```diff
@@ -65,79 +72,71 @@ def is_default(self, key):
     """
     return self.get_rank(key) in (RankedValue.NONE, RankedValue.HARDCODED)
 
-  def get(self, key, default=None):
+  def get(self, key: Key, default: Optional[Value] = None):
```

Comment:
Originally I had `-> Value`, but removed it because it resulted in MyPy complaining about call sites for `get_options()`.

Open question if it's possible to preserve the type information at option registration time into the later call sites... For now, the best we can do is don't type it so MyPy infers `Any` and we can cast at the call site.